stdlib/runtime/numbers.gr

```diff
@@ -1833,7 +1833,13 @@ export let (<<) = (x: Number, y: Number) => {
   } else {
     let xval = coerceNumberToWasmI64(x)
     let yval = coerceNumberToWasmI64(y)
-    WasmI32.toGrain(reducedInteger(WasmI64.shl(xval, yval))): Number
+    if (WasmI64.leU(WasmI64.clz(xval), yval)) {
```

Comment:
It looks like this fast path only works for positive numbers—is there any other quick check that'd also work for negative numbers?

Reply:
Heh, this actually has the opposite effect that you are imagining: `clz` of a negative number is zero, so we always promote to a big integer. If you think that's a problem, I can try to refine the condition a bit more, but I just tested this program, and it works as intended:

```grain
import BigInt from "bigint"
print(5 << 64)
print(-5 << 64)
print(BigInt.shl(-5t, 64l))
```
output:
```
92233720368547758080
-92233720368547758080
-92233720368547758080
```

Reply:
Sorry if I wasn't clear—I wasn't trying to say that this wasn't correct, just that exactly your point—`clz` of a negative number is always zero, so negative numbers never make it to the fast path. I think it'd be better to have a check that entered the fast path for both positive and negative numbers.

Reply:
@ospencer Got it. I think that the version I just pushed should do it.